src/vmm/src/guest_config/common/mod.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO - Remove allow(unused) once types have been used
+
+use std::collections::{HashMap, HashSet};
+use std::ops::Add;
+
+#[cfg(target_arch = "x86_64")]
+use crate::guest_config::cpuid::Cpuid;
+
+/// Type defined to represent registers across CPU vendors
+/// (pointers and values)
+pub trait Numeric: Copy + Add<Self, Output = Self> {}
+impl Numeric for u32 {}
+impl Numeric for u64 {}
+impl Numeric for u128 {}
+
+/// Map to find and define model-specific registers
+#[cfg(target_arch = "x86_64")]
+pub type MsrRegisterMap = HashMap<RegisterPointer<u32>, RegisterValue<u32>>;
+/// Map to find and define registers for 64-bit ARM CPUs
+#[cfg(target_arch = "aarch64")]
+pub type Aarch64RegisterMap = HashMap<RegisterPointer<u64>, RegisterValue<u128>>;
+
+/// A trait type to be customized for each CPU architecture
+/// to modify CPU configuration for the Firecracker guest
+pub trait ConfigurationModifier {
+    /// Applies the CPUTemplate against the CPUConfiguration
+    fn apply_template(&self, cpu_template: CPUTemplate) -> Box<CPUConfiguration>;
```

Comment:
I like "CpuTemplate" and "CpuConfig" more.

Reply:
It has been changed.